integration/port_forward_test.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/integration/skaffold"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubectl"
+	kubectx "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/portforward"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
+)
+
+func TestPortForward(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if ShouldRunGCPOnlyTests() {
+		t.Skip("skipping test that is not gcp only")
+	}
+
+	ns, _, deleteNs := SetupNamespace(t)
+	defer deleteNs()
+
+	dir := "examples/microservices"
+	skaffold.Run().InDir(dir).InNs(ns.Name).RunOrFailOutput(t)
+
+	cfg, err := kubectx.CurrentConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubectlCLI := kubectl.NewFromRunContext(&runcontext.RunContext{
+		KubeContext: cfg.CurrentContext,
+		Opts: config.SkaffoldOptions{
+			Namespace: ns.Name,
+		},
+	})
+
+	logrus.SetLevel(logrus.TraceLevel)
+	portforward.WhiteBoxPortForwardCycle(t, kubectlCLI, ns.Name)
+}
+
+// TestPortForwardDeletePod tests that port forwarding works
+// as expected. Then, the test force deletes a pod,
+// and tests that the pod eventually comes up at the same port again.
+func TestPortForwardDeletePod(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if !ShouldRunGCPOnlyTests() {
+		t.Skip("skipping test that is not gcp only")
+	}
+
+	ns, _, deleteNs := SetupNamespace(t)
+	defer deleteNs()
+
+	rpcAddr := randomPort()
+	env := []string{fmt.Sprintf("TEST_NS=%s", ns.Name)}
+	cmd := skaffold.Dev("--cache-artifacts=true", "--port-forward", "--rpc-port", rpcAddr, "-v=trace").InDir("examples/microservices").InNs(ns.Name).WithEnv(env)
+	stop := cmd.RunBackground(t)
+	defer stop()
+
+	client, shutdown := setupRPCClient(t, rpcAddr)
+	defer shutdown()
+
+	// create a grpc connection. Increase number of reties for helm.
+	stream, err := readEventAPIStream(client, t, 20)
+	if stream == nil {
+		t.Fatalf("error retrieving event log: %v\n", err)
+	}
+
+	// read entries from the log
+	entries := make(chan *proto.LogEntry)
+	go func() {
+		for {
+			entry, _ := stream.Recv()
+			if entry != nil {
+				entries <- entry
+			}
+		}
+	}()
+
+	localPort := getLocalPortFromPortForwardEvent(t, entries, "leeroy-app", "service")
+	assertResponseFromPort(t, localPort, constants.LeeroyAppResponse)
+
+	// now, delete all pods in this namespace.
+	kubectlCLI := getKubectlCLI(t, ns.Name)
+
+	killPodsCmd := kubectlCLI.Command(context.Background(),
+		"delete",
+		"pods", "--all",
+		"-n", ns.Name,
+	)
+
+	if output, err := killPodsCmd.CombinedOutput(); err != nil {
+		t.Fatalf("error deleting all pods: %v \n %s", err, string(output))
+	}
+	// port forwarding should come up again on the same port
+	assertResponseFromPort(t, localPort, constants.LeeroyAppResponse)
```

Comment:
Can you run this test 100 times and make sure there are no flakes. 

```
"INTEGRATION_TEST_ARGS="-test.run  ^TestPortForwardDeletePod --test.count 100" make integration

Reply:
Good idea -- for some reason my laptop gets really slow when running tests, so I ran 20 (~20 min) and it passed against Minikube.